test/unit/config.test.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const path = require('path');
+const assert = require('assert');
+const os = require('os');
+const mockFs = require('mock-fs');
+
+const {
+  getHomeDir,
+  getNcuDir,
+  getConfigPath,
+  getConfig,
+  getMergedConfig,
+  writeConfig,
+  updateConfig
+} = require('../../lib/config');
+
+const username = 'foo';
+const token = '123456789';
+const globalFileContent =
+  `{
+    "username": "${username}",
+    "token": "${token}"
+  }`;
+const localUsername = 'local';
+const localFileContent =
+  `{
+    "username": "${localUsername}"
+  }`;
+const newUsername = 'bar';
+const newToken = 'asdfghj';
+
+describe('config', () => {
+  beforeEach(() => {
+    mockFs({
```

Comment:
Does this really needs to get re-mocked before every single test? I think it doesn't need to be so can we move this outside of describe and restore at the end or instead of `beforeEach` and `afterEach` make it `before` and `after`.

Reply:
The file is changing in every `writeFile`, `writeJson`, `writeConfig` and `updateConfig`, so we need to isolate test cases.

I could put the `mockFs.restore()` in the specific `describe`, but I think that makes the code less readable.

Reply:
Ok that make sense.